completion/plugin-commands-completion/src/complete.ts

```diff
@@ -32,7 +33,9 @@ export async function complete (
   if (input.currentTypedWordType !== 'option') {
     if (input.lastOption === '--filter') {
       const workspaceDir = await findWorkspaceDir(process.cwd()) ?? process.cwd()
-      const allProjects = await findWorkspacePackages(workspaceDir, {
+      const workspaceManifest = await readWorkspaceManifest(workspaceDir)
+      const allProjects = await findWorkspacePackagesUsingPatterns(workspaceDir, {
+        patterns: workspaceManifest?.packages ?? ['.', '**'],
```

Comment:
I added `?? ['.', '**']` here since that's what `plugin-commands-completion` is currently doing, but this feels like a bug to me. If there's no `pnpm-workspace.yaml`, I would think `allProjects` should be empty?

Reply:
You may still use `pnpm i -r` even outside a workspace.